omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/DiagnosisController2_2Test.java

```diff
@@ -0,0 +1,128 @@
+public class DiagnosisController2_2Test extends MainResourceControllerTest {
+
+	private DiagnosisService diagnosisService;
+
+	private Encounter encounter;
+
+	@Before
+	public void before() throws Exception {
+		executeDataSet(RestTestConstants2_2.DIAGNOSIS_TEST_DATA_XML);
+		this.diagnosisService = Context.getDiagnosisService();
+		this.encounter = new Encounter();
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getURI()
+	 */
+	@Override
+	public String getURI() {
+		return RestConstants.VERSION_1 + "/patientdiagnoses";
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getUuid()
+	 */
+	@Override
+	public String getUuid() {
+		return RestTestConstants2_2.DIAGNOSIS_UUID;
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getAllCount()
+	 */
+	@Override
+	public long getAllCount() {
+		return diagnosisService.getPrimaryDiagnoses(this.encounter).size();
+	}
+
+	@Test
+	public void shouldCreateADiagnosis() throws Exception {
+		long originalCount = getAllCount();
+
+		SimpleObject diagnosisSource = new SimpleObject();
+		diagnosisSource.add("diagnosisId", 1);
+		diagnosisSource.add("encounter", new Encounter());
+		diagnosisSource.add("diagnosis", new CodedOrFreeText());
+		diagnosisSource.add("certainty", CONFIRMED);
+		diagnosisSource.add("rank", 1);
+
+		String json = new ObjectMapper().writeValueAsString(diagnosisSource);
+
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+
+		SimpleObject newDiagnosisSource = deserialize(handle(req));
+
+		Assert.assertNotNull(PropertyUtils.getProperty(newDiagnosisSource, "uuid"));
+		Assert.assertEquals(originalCount + 1, getAllCount());
+	}
+
+	@Test
+	public void shouldFetchExistingDiagnosis() throws Exception {
+
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + getUuid());
+		SimpleObject result = deserialize(handle(req));
+
+		Diagnosis diagnosis = diagnosisService.getDiagnosisByUuid(getUuid());
+
+		Assert.assertEquals(diagnosis.getUuid(), PropertyUtils.getProperty(result, "uuid"));
+	}
+
+	@Test
+	public void shouldVoidDiagnosis() throws Exception{
+
+		Diagnosis diagnosis = diagnosisService.getDiagnosisByUuid(getUuid());
+
+		Assert.assertFalse(diagnosis.isVoided());
+
+		MockHttpServletRequest req = request(RequestMethod.DELETE, getURI() + "/" + getUuid());
+		req.addParameter("reason", "test");
+		handle(req);
+
+		diagnosis = diagnosisService.getDiagnosisByUuid(getUuid());
+		Assert.assertTrue(diagnosis.isVoided());
+		Assert.assertEquals("test", diagnosis.getVoidReason());
+	}
+
+	@Test
+	public void shouldPurgeDiagnosis() throws Exception{
+		MockHttpServletRequest req = request(RequestMethod.DELETE, getURI() + "/" + getUuid());
+		req.addParameter("purge", "");
+		handle(req);
+
+		Assert.assertNull(diagnosisService.getDiagnosisByUuid(getUuid()));
```

Comment:
Are you sure this was not null before the purging?

Reply:
Let me add a test for the same

Reply:
Can you also add a test for unvoiding?